src/containers/offer-page/create-offer/specialization-block/SpecializationBlock.tsx

```diff
@@ -0,0 +1,115 @@
+  const { response: categoriesItems, loading: categoriesLoading } =
+    useCategoriesNames()
+
+  const { response: subjectsItems, loading: subjectsLoading } =
+    useSubjectsNames({
+      category: data.categoryId
+    })
+
+  const getValue = useCallback(
+    (array: CategoryNameInterface[], key: keyof T) =>
+      array.find((option) => option._id === data[key]) || null,
+    [data]
+  )
+
+  const handleAutocompleteChange =
+    (key: keyof Pick<T, 'categoryId' | 'subjectId'>) =>
+    (_: React.ChangeEvent, value: CategoryNameInterface | null) => {
+      handleNonInputValueChange(key, value?._id || '')
+    }
+
+  const getLabel = (option: CategoryNameInterface) => option.name || ''
+
+  const isOptionEqualToValue = (
+    option: CategoryNameInterface,
+    value: CategoryNameInterface
+  ) => option?._id === value?._id
+
+  const levelOptions = levelsTranslationKeys.map((level) => t(level))
+
+  return (
+    <OrderedListItem
+      number={1}
+      title={t(`offerPage.createOffer.title.firstStep.${userRole}`)}
+    >
+      <Box sx={styles.specialization}>
+        <Box>
+          <Typography sx={[styles.description, styles.category]}>
+            {t(`offerPage.createOffer.description.category.${userRole}`)}
+          </Typography>
+          <AppAutoComplete
+            getOptionLabel={getLabel}
+            isOptionEqualToValue={isOptionEqualToValue}
+            loading={categoriesLoading}
+            onBlur={handleBlur('categoryId')}
+            onChange={handleAutocompleteChange('categoryId')}
+            options={categoriesItems}
+            sx={styles.inputs}
+            textFieldProps={{
+              label: t('offerPage.createOffer.labels.category'),
+              error: Boolean(errors.categoryId),
+              helperText: t(errors.categoryId) || ' '
+            }}
+            value={getValue(categoriesItems, 'categoryId')}
+          />
+          <AppAutoComplete
+            disabled={!data.categoryId}
+            getOptionLabel={getLabel}
+            isOptionEqualToValue={isOptionEqualToValue}
+            loading={subjectsLoading}
+            onBlur={handleBlur('subjectId')}
+            onChange={handleAutocompleteChange('subjectId')}
+            options={subjectsItems}
+            textFieldProps={{
+              error: Boolean(data.categoryId && errors.subjectId),
+              helperText: data.categoryId ? t(errors.subjectId) : ' ',
+              label: t('offerPage.createOffer.labels.subject')
+            }}
+            value={getValue(subjectsItems, 'subjectId')}
+          />
+        </Box>
+        <Box sx={styles.inputBlock}>
+          <Typography sx={styles.description}>
+            {t(`offerPage.createOffer.description.level.${userRole}`)}
+          </Typography>
+          <CheckboxList
+            error={t(errors.proficiencyLevel)}
+            items={levelOptions}
+            onChange={(value) =>
+              handleNonInputValueChange('proficiencyLevel', value)
```

Comment:
Mowe out from jsx as you did with levelOptions

Reply:
fixed